app/uk/gov/hmrc/agentassurance/services/EntityCheckService.scala

```diff
@@ -65,15 +69,54 @@ class EntityCheckService @Inject() (
       }
     }
 
-    def entityChecks(utr: Utr): Future[Seq[EntityCheckException]] = {
+    def entityChecks(arn: Arn, utr: Utr): Future[Seq[EntityCheckException]] = {
       mongoLockService
         .dailyLock(utr) {
           val saUtr = SaUtr(utr.value)
           Future
             .sequence(Seq(deceasedStatusCheck(saUtr), refusalToDealCheck(saUtr)))
             .map(_.flatten)
         }
-        .map(_.getOrElse(Seq.empty[EntityCheckException]))
+        .map {
+          case Some(entityCheckExceptions) =>
+            val onRefusalListAgentCheckOutcomes: AgentCheckOutcome = entityCheckExceptions
+              .collect {
+                case AgentIsOnRefuseToDealList =>
+                  AgentCheckOutcome(
+                    agentCheckType = "onRefusalList",
+                    isSuccessful = false,
+                    failureReason = Some(AgentIsOnRefuseToDealList.failedChecksText)
+                  )
+              }
+              .headOption
+              .getOrElse(AgentCheckOutcome(agentCheckType = "onRefusalList", isSuccessful = true, failureReason = None))
+
+            val isDeceasedAgentCheckOutcome: AgentCheckOutcome = entityCheckExceptions
+              .collect {
+                case EntityDeceasedCheckFailed =>
+                  AgentCheckOutcome(
+                    agentCheckType = "isDeceased",
+                    isSuccessful = false,
+                    failureReason = Some(EntityDeceasedCheckFailed.failedChecksText)
+                  )
+                case x @ CitizenConnectorRequestFailed(_) =>
+                  AgentCheckOutcome(
+                    agentCheckType = "isDeceased",
+                    isSuccessful = true,
```

Comment:
This should be false?

Reply:
good question. If we can not check because connection failed to citizen details what should we report in audit ??

Reply:
"Technical problem"? Let's ask David...

Reply:
David says "Technical issue" would do...we can modify later if needed.

Reply:
but we would not want to trigger an email which a false would do